### ISSUE TITLE
\incmultigraph fails on guessing file name extension

### ISSUE THREAD

**Issue by Woi** [T-F-S/incgraph#1]
Hey Thomas,

Thanks for your packages, it does help a lot. However I observed the following (mis-)behaviour:
`\incmultigraph{example-image\n}{1,2}` fails to include `example-image?.jpg`. This is in contrast to `\incmultigraph{example-image\n.jpg}{1,2}` which includes `example-image1.jpg` and `example-image2.jpg`. Consequently you always used file extensions in your documentation.

My minimalish .tex file is:

```
\documentclass[11pt,a4paper]{scrartcl}

\usepackage[utf8]{inputenc}                   % replace by the encoding you are using
\usepackage{incgraph}                         % Loads graphicx as well
\igrset{                                      % Settings for incgraph and underlaying packages
    options={                                 % Options for \includegraphics commands
        width=\paperwidth,
        height=\paperheight,
        keepaspectratio
    },
    documentpaper
}
\DeclareGraphicsExtensions{.png,.jpg}

\begin{document}
  %\incgraph{\jobname}
  %\incmultigraph{\jobname\n}{,1,...,9}
  %\incmultigraph{exampleimage\n.jpg}{,1,2}
  \incmultigraph{exampleimage\n}{1,2}
\end{document}
```

**Comment by Woi** [comment]
Just for the file: I'm using incgraph 1.12 on Fedora 26.
```
$ LANG=C
$ dnf -C info texlive-incgraph
Last metadata expiration check: 0:34:04 ago on Fri Dec  1 15:40:40 2017.
Installed Packages
Name         : texlive-incgraph
Epoch        : 6
Version      : svn36500.1.12
Release      : 33.fc26.2
Arch         : noarch
Size         : 15 k
Source       : texlive-2016-33.20160520.fc26.1.src.rpm
Repo         : @System
From repo    : fedora
Summary      : Sophisticated graphics inclusion in a PDF document
URL          : http://tug.org/texlive/
License      : LPPL 1.3
Description  : [...]
$ 
```

**Comment by Woi** [comment]
I'm not sure if this is related or a bug on it's own:

Assuming the following statements are from `example.tex`: `\includegraphics{\jobname2}` succeeds in including  `example2.jpg`, while `\incgraph{\jobname2}` fails to do so.  Using `\jobname` for `example.jpg` works fine.

**Comment by T-F-S** [comment]
You are right with your observations. This behaviour is controlled by the `existence check` key. Currently, there are two predefined settings which can be used.

1.  `ignore on not found` is a special key which sets `existence check`.  This is also the default setting of the package. The idea is that an inclusion is ignored, if the *exact* file name is not detected. There is no guessing for extensions with this setting. This allows to include a large number of files following a numbering pattern without errors, if some of these files do not exist.

2.  `fail on not found` is another special key which sets `existence check`. Here, you get the original behaviour of `\includegraphics`, i.e. extensions are added automatically, but you also get errors, if an included file does not exist. You just have to add `\gtrset{fail on not found}` to your example.

3. Some kind of mixed behaviour could be interesting, but there is not predefined key for that. The following is a quick way to add `.png` and `.jpg` without getting errors for not existing files:

```latex
\makeatletter

\def\MyFileExists#1{%
  \IfFileExists{#1}{\expandafter\@firstoftwo}{%
  \IfFileExists{#1.png}{\expandafter\@firstoftwo}{%
  \IfFileExists{#1.jpg}{\expandafter\@firstoftwo}{%
  \expandafter\@secondoftwo}}}%
}
\igrset{existence check=\MyFileExists}

\makeatother
```

Of course, it would be more elegant to read the extensions from the settings by `\DeclareGraphicsExtensions`, but, maybe, the not-so-elegant way is enough.

**Comment by Woi** [comment]
Thanks for the replay and clarification. I'm using the following solution. Instead of
```
\DeclareGraphicsExtensions{.png,.jpg}
[...]
\incmultigraph{\jobname\n}{,1,...,9}
```

I'm using
```
\incmultigraph{\jobname\n.jpg}{,1,...,9}
\incmultigraph{\jobname\n.png}{,1,...,9}
```
which is not so bad for a small amount of extensions :)

Should I close this bug or do you like to keep it as a feature request?

**Comment by T-F-S** [comment]
Leave this open for now. I will think about some nice implementation for a third version of the extension check. Currently, I have no time for it.

If I find something nice, I will add it as a feature. Otherwise, I will close this issue later myself.

**Comment by T-F-S** [comment]
Implemented with version `1.2.0 (2021-10-20)`:
- Adding/Guessing file name extensions:
    - Option `extensions`
    - Option `extensions add`
    - Option `extensions from graphics`
    - Option `ignore on not found with extensions`
    - Option `ignore on not found with extensions*`
